xla/ffi/api/c_api.h

```diff
@@ -170,6 +170,31 @@ typedef enum {
 } XLA_FFI_DataType;
 // LINT.ThenChange(ffi_test.cc)
 
+#define _(x) case XLA_FFI_DataType_##x: return #x;
```

Comment:
At a high level I want to keep c_api.h as only structs and define functions that work with these structs in:

1. ffi/api/api.h - if they can be shared between external/internal ffi implenentations
2. ffi/api/ffi.h - for external ffi
3. ffi/ffi.h - for internal ffi

Mainly because you can use "normal" C++

Reply:
Makes sense, I added this function in `c_api.h` because I was thinking the users of the C API would benefit from a function like that too since the source of truth enum is defined at the C level. Though users can build their own.
- `c_api.h` <- data type source of truth enum is here
- `api.h` (includes `c_api.h`) <- currently no ToString here
- `ffi.h` (includes `c_api.h` and `api.h`) <- only here

Reply:
The idea is that end users should not really see (and care) about C API, XLA aims to provide a c++ wrapper that hides all the details and provides a nice header-only c++17 API, just like PJRT, we have C API monstrosity but it's wrapped into c++ classes that are easier to work with

Reply:
Hmm, then why does the C API exist?

Reply:
It's a layer that defines ABI and allows compiling xla/jaxlib/custom calls with different compilers and different C++ versions, it's nearly impossible to safely pass C++ structs/classes between shared libraries.